plugins/woo-ai/src/product-description/product-description-button-container.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useTinyEditor } from '../hooks/useTinyEditor';
+
+type WooApiMessage = {
+	role: string;
+	content: string;
+};
+
+type WooApiResponse = {
+	generated_text: string;
+	previous_messages: WooApiMessage[];
+};
+
+const generatingContentPhrases = [
+	__( 'Please wait…', 'woocommerce' ),
+	__( 'Just a little longer…', 'woocommerce' ),
+	__( 'Almost there…', 'woocommerce' ),
+];
+
+const getGeneratingContentPhrase = () =>
+	generatingContentPhrases[
+		Math.floor( Math.random() * generatingContentPhrases.length )
+	];
+
+export function DescriptionButtonContainer() {
+	const [ fetching, setFetching ] = useState( false );
+	const generatingContentPhraseInterval = useRef< number >();
+	const titleEl = useRef< HTMLInputElement >(
+		document.querySelector( '#title' )
+	);
+	const [ productTitle, setProductTitle ] = useState< string >(
+		titleEl.current?.value || ''
+	);
+	const tinyEditor = useTinyEditor();
+
+	useEffect( () => {
+		titleEl.current?.addEventListener( 'keyup', ( e ) =>
+			setProductTitle( ( e.target as HTMLInputElement ).value || '' )
+		);
+	}, [ titleEl ] );
+
+	useEffect( () => {
+		if ( fetching ) {
+			tinyEditor.setContent( getGeneratingContentPhrase() );
+			generatingContentPhraseInterval.current = setInterval(
+				() => tinyEditor.setContent( getGeneratingContentPhrase() ),
+				2000
+			);
+		} else {
+			clearInterval( generatingContentPhraseInterval.current );
+		}
+	}, [ fetching ] );
+
+	const buildPrompt = () => {
+		const instructions = [
+			`Write a product description with the following product title: ${ productTitle }.`,
+			'Use a 9th grade reading level.',
+			'Make the description 300 words or less.',
+			'Structure the description using standard HTML paragraph, strong and list tags.',
+			'Do not include a heading at the very top of the description.',
+		];
+
+		const existingListContents = ( tinyEditor.getContent() || '' )
```

Comment:
I'm experimentally injecting any bullet points from the existing product description. It's too unintuitive and fragile in it's current form, but it _does_ work to inject features and even retain then for subsequent generations.